### DISCUSSION TITLE
Hayward Tristar VS950 485 issues

### DISCUSSION THREAD

**Discussion by dakowals** [tagyoureit/nodejs-poolController#956] [RS485]
Just finished wiring and installing my panel and have everything working except for the comms with the main pump.  I have been playing around with it for 3 days and cannot figure this out.  I have 2 Sequent Micro cards installed, the BAS and IND.  I am running the 485 off of the BAS.  The pump is the only 485 connection.  I would like to know anyone who has successfully connected to this pump and what they did to make it work.  See attached for my setup.  No idea what to do.  Any help appreciated.

![Screenshot 2024-05-31 100614](https://github.com/tagyoureit/nodejs-poolController/assets/43075596/fad05a1e-24ec-4704-b5e6-9336d15a44f7)

![Screenshot 2024-05-31 094640](https://github.com/tagyoureit/nodejs-poolController/assets/43075596/a09b6249-c324-4daf-8664-52eac8c35911)
![Screenshot 2024-05-31 094502](https://github.com/tagyoureit/nodejs-poolController/assets/43075596/ce950fa4-9aac-44c5-ae36-c481579e3f26)
 
![Screenshot 2024-05-31 100352](https://github.com/tagyoureit/nodejs-poolController/assets/43075596/ff90c677-5ab5-434e-b3b6-6829957c80a4)
![20240531_094251](https://github.com/tagyoureit/nodejs-poolController/assets/43075596/215d5346-a418-4482-9bcb-9045515a8aad)
![20240531_094230](https://github.com/tagyoureit/nodejs-poolController/assets/43075596/1dda4622-bd50-4286-ba31-48114923a2c4)

**Comment by tagyoureit** [comment]
Baud rate for comms needs to be 9600.

**Reply by dakowals** [reply]
I tried 9600,8,N,2 and 9600,8,N,1.  Neither worked.  Everything I've read about the Tristar and Ecostar appears to say the they need to be at 19200,8,N,2.  Thus far, packets are sending and receiving, but none are successful.

**Reply by tagyoureit** [reply]
Oh, ya, I think those pumps are 19200,N,8,2.  They can’t be on the same bus as anything else.  The pumps won’t talk until spoken to.  But try the age old wisdom of “flipping the rs485 wires”.

**Reply by dakowals** [reply]
I have done that.  Nothing else on the line besides the pump.  I do have the line broken once through a terminal block, but the signal should be good.  Not sure what else to try.

**Reply by tagyoureit** [reply]
Check the Sequent Micro cards to make sure you have the jumpers in the right place to enable the RS485 comms.

**Reply by dakowals** [reply]
I have a 8 Relay, Building Card and an Industrial Card.  All 3 Sequent Microsystem cards have rs485 on them.  Should I use one over the other?  Does only the card I use have the jumpers installed?  Is there possibly a conflict with the 485 bus having 3 cards capable of 485 installed?

**Reply by rstrouse** [reply]
> I have a 8 Relay, Building Card and an Industrial Card. All 3 Sequent Microsystem cards have rs485 on them. Should I use one over the other? Does only the card I use have the jumpers installed? Is there possibly a conflict with the 485 bus having 3 cards capable of 485 installed?

Absolutely,  the Sequent cards use the UART on the Pi for comms and cannot be shared on RS485.

**Reply by dakowals** [reply]
So the solution would be to make sure the jumpers are only installed on ONE card.  Is there a preference of reliability between the 3 cards?

**Reply by rstrouse** [reply]
There really isn't a preference with the cards you have.  Some versions of the Mega-BAS have had issues and simply switching to the relay 8 card made all the difference.

**Reply by dakowals** [reply]
So I finally had a chance to switch the 485 bus over to the SM 8 relay card V5.3.  I made sure the jumpers were removed from the IND and BAS cards and installed in the relay card.  I have the exact same issue.  I tried switching the A/B lines again and did not help.  Not having any luck.  Any other suggestions?

**Reply by rstrouse** [reply]
Where are you connecting the RS485 on the pump?  I don't have one of these pumps but I seem to recall that there was something about  not having the display connected at the same time and the address needing to be set for the pump.  On some models this is by dip switch.

It looks like the address is set on the menu display on this model.  In section 6.6 number 10 and 11 is where this address is set and the comms are enabled.

https://hayward.com/media/akeneo_connector/asset_files/t/r/tristarvs_IS32950VS_2caf.pdf.

**Comment by dakowals** [comment]
![image](https://github.com/tagyoureit/nodejs-poolController/assets/43075596/cd2473a4-c737-4335-ab91-e2a5592fd1af)
I have connected to COMBUS 7&8, with no luck.  I am currently connected to 2&3 on the display expansion connector.  My display is currently connected.  My system does not have dip switches.  I have the SPX3400DR drive module.  The address is set exactly as the guide says.  Set to Standalone/Remote : Pool Filter.

**Reply by rstrouse** [reply]
Which address have you selected in dashPanel for the pump?

EDIT:  Try each address 1 at a time to see if it responds.  If I remember correctly, the addresses when they were on the dip switches were goofy in that the 1st address was not actually the pool filter.

**Reply by rstrouse** [reply]
A bit more on this.  Try setting the pump to Aux 1/Spa Filter on the pump.  If you look at the conversation in the issue ticket you will see that one user was using that address for address 1.  Perhaps there is more goofiness with the pump address that we don't understand.

https://github.com/tagyoureit/nodejs-poolController/issues/393

**Reply by dakowals** [reply]
I have address 1, but have tried 1-4.  I have also tried 1-4 while changing the pump address from POOL FILTER to AUX/SPA FILTER and also a few more different combinations.  I read a while ago, but cannot find now, about the SP3200DR/SPX3400DR drive module having a  rev that did not have a firmware that would communicate.  I checked mine ad is Drive: R1.23, Display: R3.1.2.  I have also tried disconnecting the display.  I will be trying a few more combinations when I get home tonight.  Even though I have a usable 485 bus on the BAS, IND and RELAY cards, I think I may purchase a separate USB 485 and try that.  Hoping for some other suggestions, this is the only thing that does not work yet.  Thanks.
